File: Javascript/8-Kyu/Powers-of-2.js
// Powers of 2
function powersOfTwo(n){

    let powerList = [];
    
    for (i = 0; i <= n; i++) {
        powerList.push(2 ** i);
    };

    return powerList;

};

// Tests 
console.log(powersOfTwo(0));  // --> [1]
console.log(powersOfTwo(1));  // --> [1, 2]
console.log(powersOfTwo(2));  // --> [1, 2, 4]

File: Python/8-Kyu/Powers-of-2.py
# Powers of 2
def powers_of_two(n):

    return [2 ** x for x in range (0, n + 1)]

# Tests
print(powers_of_two(0))  # --> [1]
print(powers_of_two(1))  # --> [1, 2]
print(powers_of_two(4))  # --> [1, 2, 4, 8, 16]
